src/main/scala/org/broadinstitute/dsde/workbench/leonardo/dao/HttpSamDAO.scala

```diff
@@ -0,0 +1,65 @@
+package org.broadinstitute.dsde.workbench.leonardo.dao
+
+import akka.actor.ActorSystem
+import akka.http.scaladsl.Http
+import akka.http.scaladsl.marshallers.sprayjson.SprayJsonSupport._
+import akka.http.scaladsl.model.HttpMethods._
+import akka.http.scaladsl.model.headers._
+import akka.http.scaladsl.model.StatusCodes._
+import akka.http.scaladsl.model._
+import akka.http.scaladsl.unmarshalling.Unmarshal
+import akka.stream.ActorMaterializer
+import org.broadinstitute.dsde.workbench.leonardo.model.{LeoException, UserInfo}
+import org.broadinstitute.dsde.workbench.model.WorkbenchIdentityJsonSupport._
+import org.broadinstitute.dsde.workbench.model.WorkbenchUserServiceAccountEmail
+import org.broadinstitute.dsde.workbench.util.health.StatusJsonSupport._
+import org.broadinstitute.dsde.workbench.util.health.SubsystemStatus
+
+import scala.concurrent.{ExecutionContext, Future}
+
+case class CallToSamFailedException(uri: Uri, status: StatusCode, msg: String)
+  extends LeoException(s"Call to Sam endpoint [${uri.path}] failed with status $status. Message: $msg", status)
+
+/**
+  * Created by rtitle on 10/16/17.
+  */
+class HttpSamDAO(val baseSamServiceURL: String)(implicit system: ActorSystem, materializer: ActorMaterializer, executionContext: ExecutionContext) extends SamDAO {
```

Comment:
I thought about putting `SamDAO/HttpSamDAO/MockSamDAO` in workbench-libs, but wasn't really sure about the right place for them. Ideally Sam would publish its own client library as a dependency. So I just left it in Leo, with the intention that this could be moved later.

Note also this isn't a complete implementation of the Sam API; just the stuff Leo needs.